lib/index.d.ts

```diff
@@ -0,0 +1,801 @@
+import type { Response } from "node-fetch"
+export { Response }
+
+export interface DinkyRequestOptions {
+  /**
+   * User's personal API key taken from their account.
+   */
+  key?: string
+
+  /**
+   * An ID of a filter found on filters page.
+   * The ID can be found on filters page: <https://derpibooru.org/filters>
+   */
+  filter?: number
+}
+
+export class Dinky {
+  /**
+   * Creates an instance of derpibooru API client.
+   * See `DinkyRequestOptions` for more details on configuration.
+   *
+   * Default options: `{}`
+   */
+  constructor(options?: DinkyRequestOptions)
+
+  /**
+   * Creates a request handler for `/api/v1/json/search`
+   *
+   * @param query – query params
+   */
+  search(...query: (string | string[])[]): ImagesSearch
+
+  /**
+   * Creates a request handler for `/api/v1/json/comments`
+   */
+  comments(): Comments
+
+  /**
+   * Creates a request handler for `/api/v1/json/tags`
+   */
+  tags(): Tags
+
+  /**
+   * Creates a request handler for `/api/v1/json/images`
+   */
+  images(): Images
+
+  /**
+   * Creates a `Lists` class that contains a bunch of shortcuts for `Search`
+   */
+  lists(): Lists
+}
+
+export function dinky(options: DinkyRequestOptions): Dinky
```

Comment:
The options argument is optional too : D

Reply:
Good catch # 2, fixed this in https://github.com/octet-stream/dinky/pull/7/commits/a8fdbf53f65e36070191c51549dbb472b11ff6a2